File: backend/src/main/kotlin/com/example/backend/models/Commande.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "commande")
class Commande (
  @Id
  @Column(name = "id", nullable = false)
  var id: Long? = null,

  @Column(name = "employe_id")
  var employeId: Int? = null,

  @Column(name = "dateCreation")
  var dateCreation: LocalDateTime? = null,

  @Column(name = "dateLivraison")
  var dateLivraison: LocalDateTime? = null,

  @Column(name = "note")
  var note: String? = null,

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "fournisseur_id")
  var fournisseur: com.example.backend.models.Fournisseur? = null,

  @OneToMany(mappedBy = "commande", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
  var produits: MutableList<ProduitCmd?> = mutableListOf(),

  @Column(name = "qtiteCmd")
  var qtiteCmd: Int? = null,

  @Column(name = "qtiteRecu")
  var qtiteRecu: Int? = null,

  @Column(name = "uniteGratuite")
  var uniteGratuite: Int? = null,

  @Column(name = "montantCmd")
  var montantCmd: Double? = 0.0,

  @Column(name = "montantRecu")
  var montantRecu: Double? = 0.0,

  @Column(name = "etat", length = 32)
  var etat: String? = null,

  @Column(name = "ref", length = 32)
  var ref: String? = null,

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
){
  // Constantes pour les états de la caisse
  companion object {
    const val COMMANDE_ANNULER = "ANNULER"
    //    const val COMMANDE_CLOTURE = "CLOTURE"
    const val COMMANDE_LIVREE = "LIVREE"
    const val COMMANDE_EN_COURS = "EN_COURS"
    const val COMMANDE_EN_ATTENTE = "EN_ATTENTE"
    const val COMMANDE_CLOTUREE = "CLOTUREE"

    const val COMMANDE_RECEPTION_TYPE_COMPLETE = "COMPLETE"
    const val COMMANDE_RECEPTION_TYPE_PARTIEL = "PARTIEL"
    const val COMMANDE_RECEPTION_TYPE_ANNULER = "ANNULER"

  }
}

File: backend_py/pharmaPython/app/models/commande.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class Commande(Base):
  __tablename__ = "commande"

  id = Column(Integer, primary_key=True, index=True)
  employe_id = Column(Integer)
  date_creation = Column(DateTime)
  date_livraison = Column(DateTime)
  note = Column(String)
  fournisseur_id = Column(Integer, ForeignKey("fournisseur.id"))
  fournisseur = relationship("Fournisseur", lazy="joined")
  qtite_cmd = Column(Integer)
  qtite_recu = Column(Integer)
  unite_gratuite = Column(Integer)
  montant_cmd = Column(Float, default=0.0)
  montant_recu = Column(Float, default=0.0)
  etat = Column(String(32))
  ref = Column(String(32))
  supprimer = Column(Integer, default=0)

  produits = relationship("ProduitCmd", back_populates="commande", lazy="joined", cascade="all, delete-orphan")

  # Constantes d’état (à importer ailleurs si nécessaire)
  COMMANDE_ANNULER = "ANNULER"
  COMMANDE_LIVREE = "LIVREE"
  COMMANDE_EN_COURS = "EN_COURS"
  COMMANDE_EN_ATTENTE = "EN_ATTENTE"
  COMMANDE_CLOTUREE = "CLOTUREE"
  RECEPTION_COMPLETE = "COMPLETE"
  RECEPTION_PARTIEL = "PARTIEL"
  RECEPTION_ANNULER = "ANNULER"
